lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -304,22 +289,22 @@ Point TruncatedDistribution::computePDFGradient(const Point & point) const
 /* Get the CDFGradient of the distribution */
 Point TruncatedDistribution::computeCDFGradient(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
 
   const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(distribution_.getParametersCollection()[0].getDimension() + finiteLowerBound_ + finiteUpperBound_);
+  if ((x <= bounds_.getLowerBound()[0]) || (x > bounds_.getUpperBound()[0])) return Point(distribution_.getParametersCollection()[0].getDimension() + bounds_.getFiniteLowerBound()[0] + bounds_.getFiniteUpperBound()[0]);
   const Point cdfGradientX(distribution_.computeCDFGradient(point));
-  const Point cdfGradientLowerBound(finiteLowerBound_ ? distribution_.computeCDFGradient(Point(1, lowerBound_)) : Point(distribution_.getParametersCollection()[0].getDimension()));
-  const Point cdfGradientUpperBound(finiteUpperBound_ ? distribution_.computeCDFGradient(Point(1, upperBound_)) : Point(distribution_.getParametersCollection()[0].getDimension()));
+  const Point cdfGradientLowerBound(bounds_.getFiniteLowerBound()[0] ? distribution_.computeCDFGradient(bounds_.getLowerBound()) : Point(distribution_.getParametersCollection()[0].getDimension()));
+  const Point cdfGradientUpperBound(bounds_.getFiniteUpperBound()[0] ? distribution_.computeCDFGradient(bounds_.getUpperBound()) : Point(distribution_.getParametersCollection()[0].getDimension()));
   const Scalar cdfPoint = distribution_.computeCDF(point);
   Point cdfGradient(normalizationFactor_ * (cdfGradientX - cdfGradientLowerBound) - (cdfPoint - cdfLowerBound_) * normalizationFactor_ * normalizationFactor_ * (cdfGradientUpperBound - cdfGradientLowerBound));
   // If the lower bound is finite, add a component to the gradient
-  if (finiteLowerBound_)
+  if (bounds_.getFiniteLowerBound()[0])
   {
     cdfGradient.add(pdfLowerBound_ * normalizationFactor_ * ((cdfPoint - cdfLowerBound_) * normalizationFactor_ - 1.0));
   }
   // If the upper bound is finite, add a component to the gradient
-  if (finiteUpperBound_)
+  if (bounds_.getFiniteUpperBound()[0])
```

Comment:
essentially the same comments as for computePDFGradient(). These methods should throw a NotYetImplementedException if dimension>1, or the correct formulas should be implemented (and written in the comments)

Reply:
ok